administrator/components/com_joomgallery/models/images.php

```diff
@@ -591,6 +592,9 @@ public function publish($cid, $publish = 1, $task = 'publish')
       }
     }
 
+    JPluginHelper::importPlugin('content');
+    $this->_mainframe->triggerEvent('onContentChangeState', array(_JOOM_OPTION.'.image', $cid, array('publish'=>$publish,'task'=>$task)));
```

Comment:
> 1. Stick stricly to the Joomla core and pass only integers of 0 and 1. This will lead to serious problems in the JG plugins since JoomGallery knows more than two states (publish, approve, feature)

Afaik that's not true. The Joomla core knows also the state "Archived" for articles and it's value is "2". Which serious problems will occur and which JG plugin is using the event  "onCategoryChangeState" at the moment? The only one I know is your provided JoomFinderPlugin, which I' am testing together with this event. Unfortunately nothing happens with indexed images of a category if I publish or unpublish the catergory.

> 2. Add other integers for other states. Like 0&1 for publish, 2&3 for approve and 4&5 for feature. This could lead to problems for core Plugins since they only check the states 0 and 1.

Yes, that is an option. You could additionally use defined CONSTANTS to make the code more readable. Why should this lead to problems in the core plugins? Could you provide an example please. Every triggered function should and afaik does check with the parameter `extension` if it should do process the trigger.  And an **override** of `FinderIndexerAdapter::translateState()` should or could also be used.

> 3. Stay at the array approach which will solve the problem for JG plugins and will lead to problems in core plugins. And life with the uglyness of passing an array to a integer argument.

Most probably this statement has a typo. Do you meant "and will lead **not** to problems in core plugins." If yes than I have to ask you, which problems of which JG plugins do you mean and why should passing a wrong type **and** a wrong value to a Joomla core plugin lead to less problems than just passing a wrong integer value?

> Any ideas how we could solve this problem...

I don't see why this couldn't be solved with integers as well. Of course you have to adapt your code in JoomFinderPlugin accordingly.

Reply:
Oke, I changed it to option B.

The interpretation of the different integers are as the following: 
0: unpublished / 1: published / 2: archived / 3: not approved / 4: approved / 5: not featured / 6: featured